calicoctl/calico_ctl/node.py

```diff
@@ -847,8 +848,13 @@ def handle_sigterm(sig, frame):
     except KeyboardInterrupt:
         # Mainline. Someone pressed Ctrl-C.
         print "Stopping Calico node..."
+        clean_exit = True
     finally:
         # Could either be this process is being killed, or output generator
         # raises an exception.
         if stop_container_on_exit:
             docker_client.stop(container)
+            # If the container is stopped due to unknown circumstances, exit
+            # with failed status code.
+            if not clean_exit:
+                sys.exit(1)
```

Comment:
I don't think that will work since `attach()` returns the output of the `calicoctl node` command whereas `wait()` returns the status code.

Turk and I chatted about this a bit earlier.  Gist was:

- We want the sys.exit to return the status of the calicoctl command, not the container's status (which `docker_client.wait()` returns).
- `docker_client.wait(container)` will always return a non-zero status on a container that has been stopped (such as with `docker_client.stop(container)`)
- KeyboardInterrupt should probably return an error status (SIGINT status code is 130) when the `calicoctl` command is stopped with ctrl-C (other unix commands have code 130 when SIGINT'd)

I think we should be running as normal if we reach the success state, but we should exit with `sys.exit(130)` if we had a keyboard interrupt and `sys.exit(1)` otherwise if we have not hit the success state.